aiosmtpd/smtp.py

```diff
@@ -907,6 +907,11 @@ async def smtp_STARTTLS(self, arg: str):
             await self.push('454 TLS not available')
             return
         await self.push('220 Ready to start TLS')
+
+        # Discard any leftover unencrypted data
+        # See https://tools.ietf.org/html/rfc3207#page-7
+        self._reader._buffer.clear()
```

Comment:
This isn't very robust as it depends on internal implementation details. Is there a beter solution we could try?

Reply:
I thought this class does not care very much about StreamReader API stability, since it already hacks on the old StreamReader instead of re-creating it, when it sets `self._reader._transport = transport` in `connection_made` (StreamReader.set_transport does not support setting the transport again once it has been set for a reason probably).

This overwriting of an internal field is not a supported use, and therefore we must make for it by doing another hacky thing, which means clearing the buffer manually. This hack can alternatively be moved towards that hack, in order to be more apparent. WDYT?

Reply:
Ah, I see. Yes, I think it'd make sense to move the code just after those lines to make it clearer.

Reply:
Done.